src/main/java/org/jabref/logic/net/URLDownload.java

```diff
@@ -231,27 +232,49 @@ public void setPostData(String postData) {
         }
     }
 
+    /**
+     * Downloads the web resource to a String. Uses UTF-8 as encoding.
+     *
+     * @return the downloaded string
+     */
+    public String asString() throws IOException {
+        return asString(StandardCharsets.UTF_8, this.openConnection());
+    }
+
     /**
      * Downloads the web resource to a String.
      *
      * @param encoding the desired String encoding
      * @return the downloaded string
      */
     public String asString(Charset encoding) throws IOException {
-        try (InputStream input = new BufferedInputStream(this.openConnection().getInputStream());
-             Writer output = new StringWriter()) {
-            copy(input, output, encoding);
-            return output.toString();
-        }
+        return asString(encoding, this.openConnection());
     }
 
     /**
-     * Downloads the web resource to a String. Uses UTF-8 as encoding.
+     * Downloads the web resource to a String from an existing connection. Uses UTF-8 as encoding.
      *
+     * @param existingConnection an existing connection
      * @return the downloaded string
      */
-    public String asString() throws IOException {
-        return asString(StandardCharsets.UTF_8);
+    public static String asString(URLConnection existingConnection) throws IOException {
```

Comment:
Is there any reason this method is static now?

Reply:
There's still a non-static `asString()` method (actually the same as the original) that was moved to line 240. This static method exists because this overloaded version works independently of any `URLDownload` object, as it only requires an open URL Connection.

This new method was created so that I could reuse an open connection to both read the request's headersm and then (through this method), the request's body, thus avoiding one additional call to the API.